src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java

```diff
@@ -124,38 +124,41 @@ public String run(@Nonnull EnvVars launchEnv, @Nonnull String image, @CheckForNu
         if (result.getStatus() == 0) {
             return result.getOut();
         } else {
-            throw new IOException(String.format("Failed to run image '%s'. Error: %s", image, result.getErr()));
+            throw new IOException(String.format("Failed to run image '%s'. Error %d: '%s','%s'",
+                    image, result.getStatus(), result.getOut(), result.getErr()));
         }
     }
 
     /**
      * Stop a container.
-     * 
-     * <p>                              
+     *
+     * <p>
      * Also removes ({@link #rm(EnvVars, String)}) the container.
-     * 
+     *
      * @param launchEnv Docker client launch environment.
      * @param containerId The container ID.
      */
     public void stop(@Nonnull EnvVars launchEnv, @Nonnull String containerId) throws IOException, InterruptedException {
         LaunchResult result = launch(launchEnv, false, "stop", "--time=1", containerId);
-        if (result.getStatus() != 0) {
-            throw new IOException(String.format("Failed to kill container '%s'.", containerId));
+        if (result.getStatus() != 0 && result.getStatus() != 143) {
+            throw new IOException(String.format("Failed to kill container '%s'. Error %d: '%s','%s'",
+                    containerId, result.getStatus(), result.getOut(), result.getErr()));
         }
         rm(launchEnv, containerId);
     }
 
     /**
      * Remove a container.
-     * 
+     *
      * @param launchEnv Docker client launch environment.
      * @param containerId The container ID.
      */
     public void rm(@Nonnull EnvVars launchEnv, @Nonnull String containerId) throws IOException, InterruptedException {
         LaunchResult result;
         result = launch(launchEnv, false, "rm", "-f", containerId);
-        if (result.getStatus() != 0) {
-            throw new IOException(String.format("Failed to rm container '%s'.", containerId));
+        if (result.getStatus() != 0 && result.getStatus() != 143) {
```

Comment:
a lot of diff noise - is this the only change? could this be amended to just show that?

Reply:
There's a few pertinent ones, each in a separate commit. There is also a whitespace-fix commit which could be merged independently or ignored as appropriate.

Reply:
possible to separate it out in that case?

Reply:
I'm not sure I understand what you want, they are already separate commits. You don't have to merge the ones you don't think value-add.

Reply:
ah sorry misread, I did mean separate PR but can see what you mean. So the material change is the status check (in 2 places), but no real way to test this right? in that case, I think this looks reasonable.

Reply:
Yeah, I couldn't come up with a reliable test case, sorry, these problems were happening randomly and infrequently.